app/src/main/java/com/sopt/geonppang/presentation/common/LoginNeededDialog.kt

```diff
@@ -0,0 +1,128 @@
+package com.sopt.geonppang.presentation.common
+
+import android.content.Intent
+import android.os.Bundle
+import android.view.View
+import androidx.fragment.app.viewModels
+import androidx.lifecycle.flowWithLifecycle
+import androidx.lifecycle.lifecycleScope
+import com.sopt.geonppang.R
+import com.sopt.geonppang.data.datasource.local.GPDataSource
+import com.sopt.geonppang.data.service.KakaoAuthService
+import com.sopt.geonppang.databinding.DialogBottomLoginNeededBinding
+import com.sopt.geonppang.presentation.MainActivity
+import com.sopt.geonppang.presentation.auth.AuthViewModel
+import com.sopt.geonppang.presentation.auth.SignActivity
+import com.sopt.geonppang.presentation.auth.SignUpActivity
+import com.sopt.geonppang.presentation.auth.SignUpNicknameActivity
+import com.sopt.geonppang.presentation.filterSetting.FilterSettingViewModel
+import com.sopt.geonppang.presentation.login.LoginActivity
+import com.sopt.geonppang.presentation.type.AuthRoleType
+import com.sopt.geonppang.presentation.type.LoginNeededType
+import com.sopt.geonppang.presentation.type.PlatformType
+import com.sopt.geonppang.presentation.type.UserRoleType
+import com.sopt.geonppang.util.AmplitudeUtils
+import com.sopt.geonppang.util.UiState
+import com.sopt.geonppang.util.binding.BindingBottomSheetDialogFragment
+import dagger.hilt.android.AndroidEntryPoint
+import kotlinx.coroutines.flow.launchIn
+import kotlinx.coroutines.flow.onEach
+import javax.inject.Inject
+
+@AndroidEntryPoint
+class LoginNeededDialog(
+    private val loginNeededType: LoginNeededType
+) :
+    BindingBottomSheetDialogFragment<DialogBottomLoginNeededBinding>(R.layout.dialog_bottom_login_needed) {
+
+    @Inject
+    lateinit var kakaoSignService: KakaoAuthService
+    private lateinit var gpDataSource: GPDataSource
+    private val authViewModel by viewModels<AuthViewModel>()
+    private val filterViewModel by viewModels<FilterSettingViewModel>()
+
+    override fun onViewCreated(view: View, savedInstanceState: Bundle?) {
+        super.onViewCreated(view, savedInstanceState)
+        binding.lifecycleOwner = this.viewLifecycleOwner
+
+        initLayout()
+        addListener()
+        collectData()
+    }
+
+    private fun initLayout() {
+        binding.loginNeedType = loginNeededType
+    }
+
+    private fun addListener() {
+        binding.ivDismiss.setOnClickListener { dismiss() }
+
+        binding.btnStartKakao.setOnClickListener {
+            kakaoSignService.startKakaoLogin(authViewModel::signUp)
+        }
+
+        binding.tvSignupWithEmail.setOnClickListener {
+            moveToSignUp()
+        }
+
+        binding.tvLoginWithEmail.setOnClickListener {
+            moveToLogIn()
+        }
+    }
+
+    private fun collectData() {
+        gpDataSource = GPDataSource(requireContext())
+        // 카카오 회원 가입, 로그인
+        authViewModel.authRoleType.flowWithLifecycle(lifecycle).onEach { role ->
```

Comment:
라이프싸이클, 라이프싸이클스코프 수정해주세욤. 프래그먼트에 해당 사항있으면 보이는대로 수정해주십쇼~~

Reply:
앗 넵 알겠습니당

Reply:
```suggestion
        authViewModel.authRoleType.flowWithLifecycle(viewLifecycleOwner.lifecycle).onEach { role ->
```

이런식으로 수정해야하는게 맞나염..?..?